lib/media/segment_reference.js

```diff
@@ -71,6 +72,38 @@ shaka.media.InitSegmentReference = class {
       return null;
     }
   }
+
+  /**
+   * Check if two initSegmentReference have all the same values.
+   * @param {?shaka.media.InitSegmentReference} reference1
+   * @param {?shaka.media.InitSegmentReference} reference2
+   * @return {boolean}
+   */
+  static equal(reference1, reference2) {
+    if (!reference1 && !reference2) {
```

Comment:
I think this would read better by having an early return followed by all the conditions at once.  You also don't need to check the size since the start and end bytes define the size.

```js
const ArrayUtils = shaka.util.ArrayUtils;
if (!reference1 || !reference2) {
  return reference1 == reference2;
} else {
  return reference1.getStartByte() == reference2.getStartByte() &&
      reference1.getEndByte() == reference2.getEndByte() &&
      ArrayUtils.equal(reference1.getUris(), reference2.getUris());
}
```

Reply:
Ok, I'll change it!